### PR TITLE
Feature/643 745 conduit earmarks

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
[App#643](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/gh/fecgov/fecfile-web-app/643)
[App#745](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/gh/fecgov/fecfile-web-app/745)
[App#1155](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/gh/fecgov/fecfile-web-app/1155)

[App PR](https://github.com/fecgov/fecfile-web-app/pull/1162)
[API PR](https://github.com/fecgov/fecfile-web-api/pull/461)
[Validator PR](https://github.com/fecgov/fecfile-validate/pull/239)

### CODE REVIEW

**Review by mjtravers — APPROVED** [review]

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 1]
```diff
@@ -0,0 +1,779 @@
...
+        "election_code",
+        "expenditure_date",
+        "expenditure_amount",
+        "expenditure_purpose_descrip",
+        "beneficiary_committee_fec_id",
+        "beneficiary_committee_name",
+        "beneficiary_candidate_fec_id",
+        "beneficiary_candidate_first_name",
+        "beneficiary_candidate_last_name",
+        "beneficiary_candidate_office",
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:30]: need:
```
 "beneficiary_candidate_state",
 "beneficiary_candidate_district"
```

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 2]
```diff
@@ -0,0 +1,779 @@
...
+            "pattern": "^[ -~]{0,200}$",
+            "examples": [
+                "John Smith & Co."
+            ],
+            "fec_spec": {
+                "FIELD_DESCRIPTION": "PAYEE ORGANIZATION NAME",
+                "TYPE": "A/N-200",
+                "REQUIRED": "X (error)",
+                "SAMPLE_DATA": "John Smith & Co.",
+                "RULE_REFERENCE": "Auto-generated from associated Schedule A IN_KIND receipt"
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:186]: Not accurate rule reference

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 3]
```diff
@@ -0,0 +1,779 @@
...
+            "minLength": 0,
+            "maxLength": 20,
+            "pattern": "^[ -~]{0,20}$",
+            "fec_spec": {
+                "FIELD_DESCRIPTION": "ELECTION OTHER DESCRIPTION",
+                "TYPE": "A/N-20",
+                "REQUIRED": "X (conditional error)",
+                "SAMPLE_DATA": null,
+                "VALUE_REFERENCE": null,
+                "RULE_REFERENCE": "Req if Item Election Code = R, S, C, E",
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:325]: `Req if Item Election Code = O`

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 4]
```diff
@@ -0,0 +1,779 @@
...
+            "type": "string",
+            "minLength": 0,
+            "maxLength": 100,
+            "fec_spec": {
+                "FIELD_DESCRIPTION": "EXPENDITURE PURPOSE DESCRIPTION",
+                "TYPE": "A/N-100",
+                "REQUIRED": "X (error)",
+                "SAMPLE_DATA": null,
+                "VALUE_REFERENCE": null,
+                "RULE_REFERENCE": "If Transaction type is (CONDUIT_EARMARK_OUT_DEPOSITED or CONDUIT_EARMARK_OUT_UNDEPOSITED) then Earmark from XX (Individual), where XX is Individual First Name Last Name from the parent Conduit Earmark Receipt on Schedule A; If Transaction type is (PAC_CONDUIT_EARMARK_OUT_DEPOSITED or PAC_CONDUIT_EARMARK_OUT_UNDEPOSITED) then Earmark from XX (Committee), where XX is Committee Name from the parent PAC Conduit Earmark Receipt on Schedule A",
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:379]: recommend the following to preserve quotes: `"If Transaction type is (CONDUIT_EARMARK_OUT_DEPOSITED or CONDUIT_EARMARK_OUT_UNDEPOSITED) then\"Earmark from XX (Individual)\", where \"XX\" is Individual First Name Last Name from the parent Conduit Earmark Receipt on Schedule A; If Transaction type is (PAC_CONDUIT_EARMARK_OUT_DEPOSITED or PAC_CONDUIT_EARMARK_OUT_UNDEPOSITED) then \"Earmark from XX (Committee)\", where \"XX\" is Committee Name from the parent PAC Conduit Earmark Receipt on Schedule A"`

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 5]
```diff
@@ -0,0 +1,779 @@
...
+            "maxLength": 9,
+            "pattern": "^(?:[PC][0-9]{8}|[HS][0-9]{1}[A-Z]{2}[0-9]{5})$",
+            "examples": [
+                "C00654323"
+            ],
+            "fec_spec": {
+                "FIELD_DESCRIPTION": "BENEFICIARY COMMITTEE FEC ID",
+                "TYPE": "A/N-9",
+                "REQUIRED": "X (error)",
+                "RULE_REFERENCE": "Auto-generated from associated Schedule A IN_KIND receipt",
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:415]: not accurate rule reference

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 6]
```diff
@@ -0,0 +1,779 @@
...
+            "maxLength": 200,
+            "pattern": "^[ -~]{0,200}$",
+            "examples": [
+                "John Smith & Co."
+            ],
+            "fec_spec": {
+                "FIELD_DESCRIPTION": "BENEFICIARY COMMITTEE NAME",
+                "TYPE": "A/N-200",
+                "REQUIRED": "X (error)",
+                "RULE_REFERENCE": "Auto-generated from associated Schedule A IN_KIND receipt",
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:433]: not accurate rule reference

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 7]
```diff
@@ -0,0 +1,779 @@
...
+                "RULE_REFERENCE": null,
+                "FIELD_FORM_ASSOCIATION": "F3 | F3X | F3P | F3L"
+            }
+        },
+        "beneficiary_candidate_state": {
+            "title": "BENEFICIARY CANDIDATE STATE",
+            "description": "",
+            "type": [
+                "string",
+                "null"
```
Comment (toddlees) [inline schema/CONDUIT_EARMARK_OUTS.json:582]: `"type": "string"` because it is a required field

**Review by toddlees — CHANGES_REQUESTED** [review]

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 2 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:186]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 3 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:325]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 4 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:379]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 5 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:415]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 6 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:433]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 8]
```diff
@@ -0,0 +1,779 @@
...
+            "title": "MEMO CODE",
+            "description": "",
+            "type": "boolean",
+            "fec_spec": {
+                "FIELD_DESCRIPTION": "MEMO CODE",
+                "TYPE": "A/N-1",
+                "REQUIRED": "X (conditional error)",
+                "SAMPLE_DATA": null,
+                "VALUE_REFERENCE": "X",
+                "RULE_REFERENCE": "X = True",
```
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:635]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 9]
```diff
@@ -0,0 +1,779 @@
...
+                    "beneficiary_candidate_state"
+                ],
+                "properties": {
+                    "beneficiary_candidate_state": {
+                        "type": "string"
+                    }
+                }
+            }
+        }
+    ]
```
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:778]: That's a biggie :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 7 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:582]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json [hunk 10]
```diff
@@ -0,0 +1,779 @@
...
+                "VALUE_REFERENCE": "AK,AL,...",
+                "RULE_REFERENCE": null,
+                "FIELD_FORM_ASSOCIATION": "F3 | F3X | F3P | F3L"
+            }
+        },
+        "beneficiary_candidate_district": {
+            "title": "BENEFICIARY CANDIDATE DISTRICT",
+            "description": "",
+            "type": [
+                "string",
```
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:605]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 1 above)
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json:30]: :+1:

File: schema/CONDUIT_EARMARK_OUTS.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (Elaine-Krauss-TCG) [reply schema/CONDUIT_EARMARK_OUTS.json]: Oops, District is only required if the candidate is a Congressman.  Senators and Presidential candidates do not have districts.

**Pull request merged**